release.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/bash
+
+set -e
+
+current_branch=$(git branch --show-current)
+git_changes=$(git status -s | wc -l)
+if [[ $git_changes != 0 ]]; then
+	if [[ $current_branch == "master" ]]; then
+		echo "You have modified the master branch."
+		echo "Please revert or move your changes to another branch before running this script."
+	else
+		echo "Please commit or stash your changes before running this script"
+	fi
+	exit 1
+fi
+
+git fetch
+ahead=$(git status -sb | grep ahead -c)
+if [[ $ahead != 0 ]]; then
+	echo "Your local master branch is ahead of the remote master branch. Exiting."
+	exit 1
+fi
+
+
+release_type=$1
+if [[ $release_type != "minor" && $release_type != "patch" && $release_type != "major" ]]; then
+	echo "Invalid argument: $release_type"
+	exit 1
+fi
+
+read -ep "Are you sure you want to cut a $release_type release? [y/N] " continue
```

Comment:
I'm not sure how I feel about a prompt here. If we wanted to build out this script with flags, and such, I might suggest a `--quiet` or a `--prompt` for this, but I think that greatly increases the complexity of a script like this. If I had to go one way or the other, I wouldn't prompt here at all.

Reply:
Since the intention is to push directly to master, I thought having some guard rails like this in place would add a layer of protection, but I can see how it could be cumbersome. I'll chew on that for a bit.

Reply:
Think of it this way: future you runs the script, and before you can be prompted, you get distracted. You come back to this terminal 2 hours later and find that you never type `y`, and commits have now gone in as a result. Depending on the actual commit strategy, that might not work, or you might overwrite good commits (e.g. if you did something completely awful like `git push -f`)

Reply:
As a middle ground to avoid that issue, how about something like:

```
echo 'You are about to push directly to the master branch'
echo 'Ctrl-C now if you want to stop this'
sleep 5
echo 'Running script'
```

Might be overkill, but giving someone who may have accidentally run the script (maybe an intern who typed `./r<tab-complete>`, or something) a chance to cancel it could save some headaches down the road.

Reply:
I'm not sure I'm concerned about _accidentally_ running the script. If this process is working, and we _accidentally_ release, is is that big of a deal?